### ISSUE TITLE
PHP 7 support

### ISSUE THREAD

**Issue by gephaest** [php-amqplib/php-amqplib#440]
Is `php-amqplib`supports PHP 7? I checked lib by [sstalle/php7cc](https://github.com/sstalle/php7cc) package and found some warnings:

```
➜  bin git:(master) ✗ ./php7cc /path/to/project/vendor/php-amqplib

File: /path/to/project/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php
> Line 302: Bitwise shift by 32 bits
$hi \ 32;
> Line 315: Bitwise shift by 32 bits
$hi \ 32;


File: /path/to/project/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPWriter.php
> Line 333: Bitwise shift by 32 bits
$n >> 32;

Checked 48 files in 1.901 seconds
```

**Comment by Nyholm** [comment]
Im running it on php 7.0.

**Comment by sobstel** [comment]
It works ok at PHP 7.0.11 for me.

**Comment by Nyholm** [comment]
@gephaest,would you mind provide a PR with a failing unit test?

**Comment by mente** [comment]
Looking at sstalle/php7cc#77 seems like php7cc had a "bug".

**Comment by hairyhum** [comment]
Addressed in sstalle/php7cc#77
